### DISCUSSION TITLE
Plugins?

### DISCUSSION THREAD

**Discussion by CodyMWilson** [mattsoulanille/NovaJS#346] [General]
https://github.com/mattsoulanille/NovaJS/blob/51c2e5895130ab36c44206059637511d08407cf8/nova/src/nova_plugin/fire_weapon_plugin.ts#L14

Out of curiosity, why are there 'plugins' and 'nova_plugins' that are being used?

**Comment by mattsoulanille** [comment]
An ecs [Plugin](https://github.com/mattsoulanille/NovaJS/blob/main/nova_ecs/plugin.ts) is just a way of organizing how to add functionality to the nova_ecs `World` (and is not related to a Nova Plug-in). When adding a plugin to the world, the world just [calls the plugin's `build` function](https://github.com/mattsoulanille/NovaJS/blob/main/nova_ecs/world.ts#L153), passing itself (the world) to the plugin for modification.

 It may have been a mistake to use a similar name, but I couldn't really think of anything else.

The [nova_plugin](https://github.com/mattsoulanille/NovaJS/tree/main/nova/src/nova_plugin) directory is the core of the engine. It includes [system_plugin.ts](https://github.com/mattsoulanille/NovaJS/blob/main/nova/src/nova_plugin/system_plugin.ts), which includes all the logic required to fly around in a star system.

The `DeltaResource` you're seeing there is used to create deltas for when a component on an entity changes (for multiplayer). In fire_weapon_plugin, the component VulnurableToPD is being marked as a component that should be synced across multiplayer clients (as opposed to something like a sprite component).
https://github.com/mattsoulanille/NovaJS/blob/51c2e5895130ab36c44206059637511d08407cf8/nova/src/nova_plugin/fire_weapon_plugin.ts#L327-L329



I'm going to write jsdoc comments for nova_ecs, so hopefully that will help make my weird code less confusing.
